album_folder.py

```diff
@@ -4,27 +4,115 @@
 import progressbar
 import os
 import sys
+import time
+from argparse import ArgumentParser
 from shutil import copyfile
+from shutil import move
+
+
+# Source_dir : passed as parameter, where your photos are located
+# output_dir : directory under where all albuns will be created
 
 def run(source_dir, output_dir):
     bar = progressbar.ProgressBar()
-    for f in bar(os.listdir(source_dir)):
-        if os.path.splitext(f)[1] == '.json':
+
+    if not os.path.exists(output_dir):
+        os.mkdir(output_dir)
+
+    # Let's print some statistics
+    num_of_missing = 0
+    num_of_albuns = 0
+    num_of_copied = 0
+    num_of_moved = 0
+    num_of_total_files = 0
+    num_of_json = 0
+
+    all_files = os.listdir(source_dir)
+    num_of_total_files = len(all_files)
+
+    for f in bar(all_files):                                       # f is each file in directory
+        (root_file, ext) = os.path.splitext(f)
+        if ext == '.json':                                                      # only .json are treated
+            num_of_json += 1
             with open(os.path.join(source_dir, f)) as data_file:
                 data = json.load(data_file)
-                for album in data['albums']:
-                    album = album.replace("/", "_")
-                    if not os.path.exists(os.path.join(output_dir, album)):
-                        os.mkdir(os.path.join(output_dir, album))
+                album_counter = 1
+
+                file_exported_name = data['uuid']                               # This is the physical exported file
+                file_original_name = os.path.basename(data['path'])             # This is the ORIGINAL file name
+                root, file_extension = os.path.splitext(file_original_name)     # get Extension of photo file
+                imagesource = os.path.join(source_dir, file_exported_name + file_extension.lower())      # the file must be located with the json file, copied from extract_photos.py
 
-                    imagesource = data['path']
-                    imagename = data['path'].split(os.sep)[-1]
-                    imagedestination = os.path.join(output_dir, album, imagename)
+                #   Some photos do no have any albums included in. So, on those cases,
+                # we have just to copy to destination dir.
+                #   Those photos whose have albuns, will be treated later
+                number_of_albums = len(data['albums'])                          # number of albums, to copy to various albums
+                if number_of_albums == 0:
+                    imagedestination = os.path.join(output_dir, file_original_name)    # destination without album
+                    if os.path.exists(imagesource):  # let's not consider the photo was exported or will raise an exception
+                        if args.verbose:
```

Comment:
Pass `args.verbose` as a parameter to `run`. Instead of having the `if verbose` statement everywhere, define a function `vprint` (inside `run`, at line 17):
```python
def vprint(x):
    if verbose:
        print(x)
```

Reply:
Good. done.